### PR TITLE
Refactor the admin route

### PR DESCRIPTION
**Pull request by Rawan96**
#### In this section:

* I Converted the promises to async and await in the back and front end route.

Relates #144

### CODE REVIEW

File: client/src/components/container/admin/controlPage/MainPage/index.js [hunk 1]
```diff
@@ -17,19 +17,25 @@ class Dashboard extends Component {
...
+      if (error.response) {
+        adminError = error.response.data.error;
+      } else {
+        adminError = 'Failed to get the admin data!';
+      }
+      this.setState({ adminError: 'Failed to get the admin data' });
+    }
   }
 
   logout = () => {
```
Comment (AhmedSafi97) [inline client/src/components/container/admin/controlPage/MainPage/index.js:36]: I think you missed turning this into async/await syntax :sweat_smile:

File: client/src/components/container/admin/controlPage/MainPage/index.js [hunk 2]
```diff
@@ -17,19 +17,25 @@ class Dashboard extends Component {
...
-    Axios.get('/api/v1/logout')
+    axios
+      .get('/api/v1/logout')
       .then((res) => {
```
Comment (AhmedSafi97) [inline client/src/components/container/admin/controlPage/MainPage/index.js:39]: I think you missed turning this into async/await syntax :sweat_smile:

Comment (AhmedSafi97) [inline client/src/components/container/admin/controlPage/MainPage/index.js:39]: I think you missed turning this into async/await syntax :sweat_smile:

File: client/src/components/container/admin/controlPage/MainPage/index.js [hunk 3]
```diff
@@ -17,29 +17,41 @@ class Dashboard extends Component {
...
+      const { name, img } = data[0];
+      this.setState({ name, img });
+    } catch (error) {
+      let adminError;
+      if (error.response) {
+        adminError = error.response.data.error;
+      } else {
+        adminError = 'Failed to get the admin data!';
+      }
+      this.setState({ adminError: 'Failed to get the admin data' });
```
Comment (AlaaTaima) [inline client/src/components/container/admin/controlPage/MainPage/index.js:32]:
```suggestion
    } catch (error) {
      let errorMsg;
      if (error.response) {
        errorMsg = error.message;
      } else {
        errorMsg = 'Failed to get the admin data!';
      }
      this.setState({ adminError : errorMsg });
```

File: client/src/components/container/admin/controlPage/MainPage/index.js [hunk 4]
```diff
@@ -17,29 +17,41 @@ class Dashboard extends Component {
...
+    } catch(error){
+      let err
+      if (error.response) {
+        err = error.response.data.error;
+      } else {
+        err = 'Error with logout process!';
+      }
+      this.setState({ err: ('Error with logout process!')});
+
+  }
```
Comment (AlaaTaima) [inline client/src/components/container/admin/controlPage/MainPage/index.js:53]:
```suggestion
    } catch(error){
      message.error('error with logout process');
  }
```

File: client/src/components/container/admin/controlPage/MainPage/index.js [hunk 5]
```diff
@@ -17,29 +17,41 @@ class Dashboard extends Component {
...
-        } = this.props;
-        push('/admin');
-      })
-      .catch((err) => {
-        message.error('error with logout process');
-      });
-  };
+  logout = async() => {
+    try{
+      const res =  await axios.get('/api/v1/logout');
```
Comment (AlaaTaima) [inline client/src/components/container/admin/controlPage/MainPage/index.js:38]:
```suggestion
      const {data} =  await axios.get('/api/v1/logout');
      message.success(data.message, 5)
```

**Pull request merged**